hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/listener/ClientListenerServiceImpl.java

```diff
@@ -331,16 +334,17 @@ private Boolean deregisterListenerInternal(@Nullable UUID userRegistrationId) {
             }
             ClientInvocation clientInvocation = new ClientInvocation(client, request, null, subscriber);
             clientInvocation.setInvocationTimeoutMillis(Long.MAX_VALUE);
-            clientInvocation.invokeUrgent().exceptionally(throwable -> {
+            list.add(clientInvocation.invokeUrgent().exceptionally(throwable -> {
                 if (!(throwable instanceof HazelcastClientNotActiveException
                         || throwable instanceof IOException
                         || throwable instanceof TargetDisconnectedException)) {
                     logger.warning("Deregistration of listener with ID " + userRegistrationId
                             + " has failed for address " + subscriber.getRemoteAddress(), throwable);
                 }
                 return null;
-            }).joinInternal();
+            }));
         }
+        CompletableFuture.allOf(list.toArray(new CompletableFuture[registrations.size()])).join();
```

Comment:
Minor: I think, instead of an ArrayList, we could use an array and that way we can get rid of unnecessary copies and allocations